tests/sentry/workflow_engine/handlers/condition/test_group_state_handlers.py

```diff
@@ -207,3 +208,82 @@ def test_dual_write(self):
         assert dc.comparison is True
         assert dc.condition_result is True
         assert dc.condition_group == dcg
+
+
+class TestNewHighPriorityIssueCondition(ConditionTestCase):
+    condition = Condition.NEW_HIGH_PRIORITY_ISSUE
+    rule_cls = NewHighPriorityIssueCondition
+    payload = {"id": NewHighPriorityIssueCondition.id}
+
+    def setUp(self):
+        super().setUp()
+        self.job = WorkflowJob(
+            {
+                "event": self.group_event,
+                "group_state": GroupState(
+                    {
+                        "id": 1,
+                        "is_regression": True,
+                        "is_new": True,
+                        "is_new_group_environment": True,
+                    }
+                ),
+                "workflow": Workflow(environment_id=1),
+            }
+        )
+        self.dc = self.create_data_condition(
+            type=self.condition,
+            comparison=True,
+            condition_result=True,
+        )
+        self.group_event.group.priority = PriorityLevel.HIGH
```

Comment:
should we use `update()` for all of these? i feel like it would be better if the DB state reflected the class state

Reply:
like `self.group_event.group.update(priority=PriorityLevel.HIGH)`